mautic/mautic
# Fix hourly email stats chart

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ Y ]
| New feature/enhancement? (use the a.x branch)      | [ N ]
| Deprecations?                          | [ N ]
| BC breaks? (use the c.x branch)        | [ N ]
| Automated tests included?              | [ Y ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Steps to reproduce the bug:
1. Create an email and send it to the selected contacts
2. Go to the email view and select a period for the statistics no longer than 1 day (so that an hourly scale appears).
![image](https://user-images.githubusercontent.com/8580942/213145790-b62a4ff0-a6a8-43ee-82eb-581ba6bacc7f.png)
4. Statistics of sent emails will always be assigned to time 0 of a given day. Other data should display correctly.

This PR modifies the database query when the statistics scale is hourly. After its implementation, the problem should not occur.



#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Follow the steps 1-2 to reproduce this issue
3. Chart should be rendered correctly
